File: 78-subsets.rs
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut combo = vec![];
        let mut subset = vec![];
        Solution::cts(nums,&mut combo,subset,0);
        combo
    }
    fn create_subsets(nums: Vec<i32>,result:&mut Vec<Vec<i32>>,mut subset: Vec<i32>, index:i32){
        result.push(subset.clone());
        for i in (index as usize)..nums.len(){
            subset.push(nums[i]);
            Solution::create_subsets(nums.clone(), result, subset.clone(), (i+1) as i32);
            subset.pop();
        }
    }
}

File: 78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ret = []
        subset = []

        def dfs(i):
            # base case
            if(i>=len(nums)):
                ret.append(subset.copy())
                return

            # include
            subset.append(nums[i])
            dfs(i+1)

            # exclude
            subset.pop()
            dfs(i+1)

        dfs(0)
        return ret
